pkg/liqoctl/purge/handler.go

```diff
@@ -0,0 +1,518 @@
+			}
+
+			peeringPhase := foreignclusterutils.GetPeeringPhase(fc)
+			if peeringPhase == consts.PeeringPhaseNone || peeringPhase == consts.PeeringPhaseAuthenticated {
+				s.Success("Unpeer completed")
+				return
+			}
+		}
+	}
+}
+
+func (h *clusterHandler) deleteTenantNamespace(ctx context.Context) error {
+	if !h.hasClientSetup {
+		return nil
+	}
+
+	s, err := h.printer.Spinner.Start("Deleting tenant namespace")
+	utilruntime.Must(err)
+
+	tenantNamespaceManager := tenantnamespace.NewTenantNamespaceManager(h.nativeCl)
+	tenantNamespace, err := tenantNamespaceManager.GetNamespace(h.remoteClusterIdentity)
+	if client.IgnoreNotFound(err) != nil {
+		s.Fail(fmt.Sprintf("Failed to delete tenant namespace: %v", err))
+		return err
+	} else if apierrors.IsNotFound(err) {
+		s.Success("Tenant namespace already deleted")
+		return nil
+	}
+	namespace := tenantNamespace.GetName()
+
+	if err := deleteNamespaceMaps(ctx, h.cl, namespace, s); err != nil {
+		s.Fail(fmt.Sprintf("Failed to delete namespace maps: %v", err))
+		return err
+	}
+
+	if err := deleteNetworkConfigs(ctx, h.cl, namespace, s); err != nil {
+		s.Fail(fmt.Sprintf("Failed to delete network configs: %v", err))
+		return err
+	}
+
+	if err := deleteResourceRequests(ctx, h.cl, namespace, s); err != nil {
+		s.Fail(fmt.Sprintf("Failed to delete resource requests: %v", err))
+		return err
+	}
+
+	if err := deleteResourceOffers(ctx, h.cl, namespace, s); err != nil {
+		s.Fail(fmt.Sprintf("Failed to delete resource offers: %v", err))
+		return err
+	}
+
+	if err := h.cl.Delete(ctx, &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}); err != nil {
+		s.Fail(fmt.Sprintf("Failed to delete tenant namespace %s: %v", namespace, err))
+		return err
+	}
+
+	s.Success(fmt.Sprintf("Deleted tenant namespace %s", namespace))
+	return nil
+}
+
+func (h *clusterHandler) deleteNode(ctx context.Context) error {
+	if !h.hasClientSetup {
+		return nil
+	}
+
+	s, err := h.printer.Spinner.Start("Deleting virtual node")
+	utilruntime.Must(err)
+
+	if err := h.cl.DeleteAllOf(ctx, &corev1.Node{},
+		client.MatchingLabels{consts.RemoteClusterID: h.remoteClusterIdentity.ClusterID}); err != nil {
+		s.Fail(fmt.Sprintf("Failed to delete virtual node: %v", err))
+		return err
+	}
+
+	var nodeList corev1.NodeList
+	if err := h.cl.List(ctx, &nodeList, client.MatchingLabels{
+		consts.RemoteClusterID: h.remoteClusterIdentity.ClusterID}); err != nil {
+		s.Fail(fmt.Sprintf("Failed to list nodes: %v", err))
+		return err
+	}
+
+	for i := range nodeList.Items {
+		node := &nodeList.Items[i]
+		node.Finalizers = []string{}
+		if err := client.IgnoreNotFound(h.cl.Update(ctx, node)); err != nil {
+			s.WarningPrinter.Println(fmt.Sprintf("Failed to remove finalizers from node %s: %v", node.Name, err))
+			continue
+		}
+	}
+
+	s.Success("Deleted virtual node")
+	return nil
+}
+
+func deleteNetworkConfigs(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
```

Comment:
Once released this version, we should try to update go to 1.18, and check whether we could reduce some code duplication by means of generics.